var/spack/repos/builtin/packages/http-post/package.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class HttpPost(MakefilePackage):
+    """Http_post does a POST operation to an HTTP URL and dumps the results
+    to stdout. It does not do gopher, ftp, file, news, or any other type of
+    URL, only HTTP. It can be configured to do HTTPS POSTs as well."""
+
+    homepage = "http://www.acme.com/software/http_post/"
+    url      = "http://www.acme.com/software/http_post/http_post_18May2018.tar.gz"
+
+    version('18May2018', sha256='981c62bcc5cd12b8531f887b3e3779a63a7b7f370062575cded412865a20ea2c')
```

Comment:
This will be problematic if anyone ever adds another version, as the version won't sort correctly. I would change the version of `2018-05-18` and either manually specify the URL for each version or use Python's datetime module to convert from one format to another in `url_for_version`. This is a suggestion, not a requirement, it could wait until someone actually needs another version.